src/logging.cpp

```diff
@@ -249,11 +336,32 @@ namespace BCLog {
     }
 } // namespace BCLog
 
-void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const std::string& source_file, const int source_line)
+void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const std::string& source_file, const int source_line, const BCLog::LogFlags category, const BCLog::Level level)
 {
     StdLockGuard scoped_lock(m_cs);
     std::string str_prefixed = LogEscapeMessage(str);
 
+    if ((category != LogFlags::NONE || level != Level::None) && m_started_new_line) {
```

Comment:
pico-nit, maybe something like
```diff
     StdLockGuard scoped_lock(m_cs);
     std::string str_prefixed = LogEscapeMessage(str);
+    const bool both_not_none{category != LogFlags::NONE && level != Level::None};
 
-    if ((category != LogFlags::NONE || level != Level::None) && m_started_new_line) {
+    if (both_not_none && m_started_new_line) {
 
         if (category != LogFlags::NONE) {
             s += LogCategoryToStr(category);
         }
 
-        if (category != LogFlags::NONE && level != Level::None) {
+        if (both_not_none) {
             // Only add separator if both flag and level are not NONE
             s += ":";
         }
```

Reply:
I don't think I can cleanly refractor the conditions as they are not the same. The first `if` checks that either category is not none or level is not none but the second `if` checks that both category and level are not none.

Reply:
Oops! was blind there, sorry 👓